machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,154 @@
+	// if the processorType is selected as dedicated, then processors value cannot be fractional
+	// Defaults to shared.
+	// +kubebuilder:validation:Enum:="dedicated";"shared";"capped"
+	// +optional
+	ProcessorType string `json:"processorType,omitempty"`
+
+	// processors is the number of virtual processors in a virtual machine.
+	// when the processorType is selected as dedicated the processors value cannot be fractional
+	// The minimum processors is 0.5
+	// Defaults to 0.5.
+	// +optional
+	Processors string `json:"processors,omitempty"`
+
+	// memoryGiB is the size of a virtual machine's memory, in GB.
+	// The minimum memory is 32GB
+	// Defaults to 32 GB.
+	// +optional
+	MemoryGiB int32 `json:"memoryGiB,omitempty"`
+}
+
+// PowerVSResource is a reference to a specific PowerVS resource by ID, Name or RegEx
+// Only one of ID, Name or RegEx may be specified. Specifying more than one will result in
+// a validation error.
+// +union
+type PowerVSResource struct {
+	// Type identifies the resource type for this entry.
+	// Valid values are ID, Name and RegEx
+	// +kubebuilder:validation:Enum:=ID;Name;RegEx
+	// +optional
+	Type PowerVSResourceType `json:"type,omitempty"`
+	// ID of resource
+	// +optional
+	ID *string `json:"id,omitempty"`
+	// Name of resource
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// Regex to find resource
+	// Regex contains the pattern to match to find a resource
+	// +optional
+	RegEx *string `json:"regex,omitempty"`
+}
+
+// PowerVSMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It contains PowerVS-specific status information.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PowerVSMachineProviderStatus struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// instanceId is the instance ID of the machine created in PowerVS
+	// instanceId uniquely identifies a Power VS server instance(VM) under a Power VS service.
+	// This will help in updating or deleting a VM in Power VS Cloud
+	// +optional
+	InstanceID *string `json:"instanceId,omitempty"`
+
+	// serviceInstanceID is the reference to the Power VS ServiceInstance on which the machine instance will be created.
+	// serviceInstanceID uniquely identifies the Power VS service
+	// By setting serviceInstanceID it will become easy and efficient to fetch a server instance(VM) within Power VS Cloud.
+	// +optional
+	ServiceInstanceID *string `json:"serviceInstanceID,omitempty"`
+
+	// instanceState is the state of the PowerVS instance for this machine
+	// Possible instance states are ACTIVE, BUILD, SHUTOFF, REBOOT
```

Comment:
Can we make the states conform to Kube pascal case conventions?
```suggestion
	// Possible instance states are Active, Build, ShutOff, Reboot
```

Reply:
Again these are state what we get from the APIs, do we really need to change the case here?